src/packagedef.jl

```diff
@@ -640,16 +641,19 @@ function handle_deletions(pkgdata, file)
     fi = maybe_parse_from_cache!(pkgdata, file)
     maybe_extract_sigs!(fi)
     mexsold = fi.modexsigs
-    filep = normpath(joinpath(basedir(pkgdata), file))
+    idx = fileindex(pkgdata, file)
+    filep = pkgdata.info.files[idx]
+    if isa(filep, AbstractString)
+        filep = normpath(joinpath(basedir(pkgdata), file))
+    end
     topmod = first(keys(mexsold))
-    fileok = file_exists(filep)
+    fileok = file_exists(String(filep))
```

Comment:
If you're doing it this way, any reason not to use `string` instead of `String`? Conversely, to insulate this from invalidation, `String(filep)::String` might be necessary.